src/indexer/delete_queue.rs

```diff
@@ -53,7 +53,7 @@ impl DeleteQueue {
             return block;
         }
         let block = Arc::new(Block {
-            operations: Arc::default(),
+            operations: Arc::new([DeleteOperation::default(); 0]),
```

Comment:
Can you change it to `[]` ?
I like that your version is explicity about the content of the array, but I have no clue on whether it DeleteOption::default() is effectively called or not at all.